src/main/kotlin/org/rust/ide/actions/ShareInPlaygroundAction.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.google.common.annotations.VisibleForTesting
+import com.google.gson.Gson
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.notification.NotificationAction
+import com.intellij.notification.NotificationListener
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.application.runReadAction
+import com.intellij.openapi.ide.CopyPasteManager
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.util.io.HttpRequests
+import org.jetbrains.annotations.TestOnly
+import org.rust.RsBundle
+import org.rust.cargo.project.workspace.CargoWorkspace
+import org.rust.ide.notifications.showBalloon
+import org.rust.ide.utils.USER_AGENT
+import org.rust.lang.core.psi.RsFile
+import org.rust.openapiext.JsonUtils
+import java.awt.datatransfer.StringSelection
+
+class ShareInPlaygroundAction : DumbAwareAction() {
+
+    override fun update(e: AnActionEvent) {
+        val file = e.getData(CommonDataKeys.PSI_FILE) as? RsFile
+        e.presentation.isEnabledAndVisible = file != null
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        val file = e.getData(CommonDataKeys.PSI_FILE) as? RsFile ?: return
+        val editor = e.getData(CommonDataKeys.EDITOR)
+
+        val (text, hasSelection) = runReadAction {
+            val selectedText = editor?.selectionModel?.selectedText
+            if (selectedText != null) {
+                selectedText to true
+            } else {
+                file.text to false
+            }
+        }
+
+        if (!confirmShare(file, hasSelection)) return
+
+        val channel = file.cargoProject?.rustcInfo?.version?.channel?.channel ?: "stable"
+        val edition = (file.crate?.edition ?: CargoWorkspace.Edition.EDITION_2018).presentation
+
+        object : Task.Backgroundable(project, RsBundle.message("action.Rust.ShareInPlayground.progress.title")) {
+
+            @Volatile
+            private var gistId: String? = null
+
+            override fun shouldStartInBackground(): Boolean = true
+            override fun run(indicator: ProgressIndicator) {
+                val json = Gson().toJson(PlaygroundCode(text))
+                val response = HttpRequests.post("$playgroundHost/meta/gist/", "application/json")
+                    .userAgent(USER_AGENT)
+                    .connect {
```

Comment:
Do you intentionally ignore `IOException` here and use `parseJsonObject` instead of `tryParseJsonObject` below?

Reply:
Yep. In case of exception, platform call `onThrowable` and we log exception and show error notification to user